### PR TITLE
Enable reduce_indices over AbstractArrays

### LINKED ISSUE #193: mapslices errors for AxisArrays
Trying to call `mapslices` over an `AxisArray` errors which I think can be fixed with the appropriate method for `reduce_indices`.

This seems somewhat related to #7.

MWE:
```julia
julia> A = AxisArray(collect(reshape(1:15,3,5)), :y, :x)
2-dimensional AxisArray{Int64,2,...} with axes:
    :y, Base.OneTo(3)
    :x, Base.OneTo(5)
And data, a 3×5 Array{Int64,2}:
 1  4  7  10  13
 2  5  8  11  14
 3  6  9  12  15

julia> mapslices(sum, A; dims=1)
ERROR: MethodError: no method matching reduced_indices(::Tuple{Axis{:y,UnitRange{Int64}}}, ::UnitRange{Int64})
Closest candidates are:
  reduced_indices(::Tuple{Vararg{Axis,N} where N}, ::Tuple{}) at /Users/glenn/.julia/dev/AxisArrays/src/core.jl:350
  reduced_indices(::Tuple{Vararg{Axis,N} where N}, ::Integer) at /Users/glenn/.julia/dev/AxisArrays/src/core.jl:352
  reduced_indices(::Tuple{Vararg{Axis,N}}, ::Tuple{Vararg{Int64,N}} where N) where N at /Users/glenn/.julia/dev/AxisArrays/src/core.jl:357
  ...
Stacktrace:
 [1] reduced_indices(::AxisArray{Int64,1,Array{Int64,1},Tuple{Axis{:y,UnitRange{Int64}}}}, ::UnitRange{Int64}) at /Users/glenn/.julia/dev/AxisArrays/src/core.jl:347
 [2] mapslices(::typeof(sum), ::AxisArray{Int64,2,Array{Int64,2},Tuple{Axis{:y,Base.OneTo{Int64}},Axis{:x,Base.OneTo{Int64}}}}; dims::Int64) at ./abstractarray.jl:2098
 [3] top-level scope at REPL[90]:1
```

### PR DESCRIPTION
**Pull request by glennmoy**
Somewhat closes #193 as it only accepts integer `dims`.

```julia
julia> A = AxisArray(collect(reshape(1:15,3,5)), :y, :x)
2-dimensional AxisArray{Int64,2,...} with axes:
    :y, Base.OneTo(3)
    :x, Base.OneTo(5)
And data, a 3×5 Array{Int64,2}:
 1  4  7  10  13
 2  5  8  11  14
 3  6  9  12  15

julia> mapslices(sum, A; dims=1)
1×5 Array{Int64,2}:
 6  15  24  33  42
```

I'm not fully sure what's needed to enable `Axis` `dims`, perhaps overloading `Base.mapslices` directly since it seems to assume the `dims` args are always integers (used for indexing).
```julia
julia> mapslices(sum, A; dims=Axis{:y})
ERROR: ArgumentError: invalid index: Axis{:y,T} where T of type UnionAll
Stacktrace:
 [1] to_index(::Type{T} where T) at ./indices.jl:297
 [2] to_index(::Array{Any,1}, ::Type{T} where T) at ./indices.jl:274
 [3] to_indices at ./indices.jl:325 [inlined]
 [4] to_indices at ./indices.jl:322 [inlined]
 [5] setindex!(::Array{Any,1}, ::Base.Slice{Base.OneTo{Int64}}, ::Type{T} where T) at ./abstractarray.jl:1153
 [6] mapslices(::typeof(sum), ::AxisArray{Int64,2,Array{Int64,2},Tuple{Axis{:y,Base.OneTo{Int64}},Axis{:x,Base.OneTo{Int64}}}}; dims::Type{T} where T) at ./abstractarray.jl:2078
 [7] top-level scope at REPL[203]:1
```

### CODE REVIEW

File: Manifest.toml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# This file is machine-generated - editing it directly is not advised
```
Comment (rofinn) [inline Manifest.toml:1]: I don't think you wanted to include the Manifest file here.

**Review by rofinn — COMMENTED** [review]
Yeah, I think you'll need to overload `mapslices` directly.

File: Manifest.toml (hunk 1 above)
Comment (glennmoy) [reply Manifest.toml:1]: yeah I don't know why that's there - it's not in the .gitignore file?

Comment (rofinn) [reply Manifest.toml]: Probably not cause this package pre-dates Julia 1.0 and Manifest.toml files.

**Comment by glennmoy** [comment]
> Yeah, I think you'll need to overload `mapslices` directly.

https://github.com/JuliaArrays/AxisArrays.jl/pull/195

**Pull request closed without merging**